### DISCUSSION TITLE
Bug: Inconsistency of star-based sorting

### DISCUSSION THREAD

**Discussion by junwha** [community/community#84715] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi, someone found that there is an inconsistency of star-based sorting when we use `https://api.github.com/search`
https://github.com/EvanLi/Github-Ranking/issues/9

The result varies across results. some popular repos are excluded in each result.

and not only the API, when I use the advanced search in Github web, the same issue occurs.

could you fix this issue?

**Comment by Naimulnobel** [comment]
The inconsistency you're experiencing might be due to the way GitHub's search API works. Here are a few things to consider:

1. **Indexing Time**: GitHub's search results are based on a continuously updated index. This means that at any given time, the search results you see might be slightly out of date compared to the live state of the repositories.

2. **Rate Limiting**: The Search API has a [custom rate limit](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#search-rate-limit), separate from the rate limit governing the rest of the REST API. For requests using Basic Authentication, OAuth, or client ID and secret, you can make up to 30 requests per minute. For unauthenticated requests, the rate limit allows you to make up to 10 requests per minute.

3. **Query Limitations**: The Search API has some limitations on how queries can be formed. For example, queries longer than 256 characters are not accepted.

4. **Result Set Limitations**: Only the first 1,000 search results are available.

If you're still experiencing inconsistencies, it might be worth reaching out to GitHub support for more specific assistance.

**Comment by ghostinhershell** [comment]
Hi @junwha0511, 

Thanks so much for reaching out in our community. I do see that your question here was addressed in the support ticket that you opened so I will go ahead and close this discussion. Let us know if you have any additional questions. Thanks!
